lib/unleash/streaming_client.rb

```diff
@@ -0,0 +1,107 @@
+require 'json'
+
+module Unleash
+  class StreamingClient
+    attr_accessor :event_source, :toggle_engine, :running, :mutex
+
+    def initialize(toggle_engine)
+      self.toggle_engine = toggle_engine
+      self.event_source = nil
+      self.running = false
+      self.mutex = Mutex.new
+    end
+
+    def start
+      self.mutex.synchronize do
```

Comment:
similar synchronization pattern to toggle_fetcher so that we guard running property

Reply:
Why do we need to guard running? That should only ever be a problem if two threads are calling start and stop on the same instance in quick succession, surely?

Reply:
No mutex needed in the scheduler for SSEs - lifecycle methods called from single thread. But in the event processor event processing happens from EventSource background threads that could potentially be concurrent so keeping it there